File: WdtcCore/src/main/java/org/wdt/wdtc/manger/FileManger.java
package org.wdt.wdtc.manger;

import java.io.File;

public class FileManger {



    public static File getMinecraftComSkin() {
        return new File(System.getProperty("user.home"), "AppData/Roaming/.minecraft/assets/skins");
    }



    public static File getWdtcConfig() {
        return new File(VMManger.getWdtcConfigFromVM(), ".wdtc");
    }


    public static File getStarterBat() {
        return new File(getWdtcCache(), "WdtcGameLauncherScript.bat");
    }

    public static File getAuthlibInjector() {
        return new File(getWdtcImplementationPath(), "authlib-injector.jar");
    }

    public static File getUsersJson() {
        return new File(getWdtcUser(), "user.json");
    }

    public static File getLlbmpipeLoader() {
        return new File(getWdtcImplementationPath(), "llvmpipe-loader.jar");
    }

    public static File getWdtcCache() {
        return new File(getWdtcConfig(), "cache");
    }

    public static File getWdtcImplementationPath() {
        return new File(getWdtcConfig(), "dependencies");
    }

    public static File getWtdcOpenJFXPath() {
        return new File(getWdtcImplementationPath(), "openjfx");
    }

    public static File getWdtcUser() {
        return new File(getWdtcConfig(), "users");
    }

    public static File getUserAsste() {
        return new File(getWdtcUser(), "assets");
    }

    public static File getUserListFile() {
        return new File(getWdtcUser(), "users.json");
    }

    public static File getSettingFile() {
        return new File(getWdtcConfig(), "setting/setting.json");
    }

    public static File getVersionManifestFile() {
        return new File(getWdtcCache(), "versionManifest.json");
    }
}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/manger/FileManger.kt
@file:JvmName("FileManger")

package org.wdt.wdtc.core.manger

import java.io.File

val minecraftComSkin: File = File(System.getProperty("user.home"), "AppData/Roaming/.minecraft/assets/skins")

val wdtcConfig: File = VMManger.wdtcConfigFromVM.resolve(".wdtc")

val wdtcSetting: File = wdtcConfig.resolve("setting")

val wdtcCache: File = wdtcConfig.resolve("cache")

val starterBat: File = wdtcCache.resolve("WdtcGameLauncherScript.bat")

val wdtcDependenciesDirectory: File = wdtcConfig.resolve("dependencies")

val wtdcOpenJFXDirectory: File = wdtcDependenciesDirectory.resolve("openjfx")

val wdtcUser: File = wdtcConfig.resolve("users")

val authlibInjector: File = wdtcDependenciesDirectory.resolve("authlib-injector.jar")

val userJson: File = wdtcUser.resolve("user.json")

val llbmpipeLoader: File = wdtcDependenciesDirectory.resolve("llvmpipe-loader.jar")

val userAsste: File = wdtcUser.resolve("assets")

val userListFile: File = wdtcUser.resolve("users.json")

val settingFile: File = wdtcSetting.resolve("setting.json")

val versionManifestFile: File = wdtcCache.resolve("versionManifest.json")

val tipsFile: File = wdtcConfig.resolve("assets").resolve("tips.txt")

val versionsJson: File = wdtcSetting.resolve("versions.json")
// TODO Plugin functionality
val pluginsDirectory: File = wdtcConfig.resolve("plugins")


